politeiawww/www.go

```diff
@@ -220,54 +266,40 @@ func (p *politeiawww) handleVerifyNewUser(w http.ResponseWriter, r *http.Request
 		query := r.URL.Query()
 		email, emailOk := query["email"]
 		token, tokenOk := query["verificationtoken"]
-		if !emailOk || !tokenOk {
-			log.Errorf("handleVerifyNewUser: Unmarshal %v", err)
-			http.Redirect(w, r, routePrefix+v1.RouteVerifyNewUserFailure,
-				http.StatusMovedPermanently)
+		sig, sigOk := query["signature"]
+		if !emailOk || !tokenOk || !sigOk {
+			RespondWithError(w, r, 0, "could not decode URL",
+				v1.UserError{
+					ErrorCode: v1.ErrorStatusInvalidInput,
+				})
 			return
 		}
 
 		vnu.Email = email[0]
 		vnu.VerificationToken = token[0]
+		vnu.Signature = sig[0]
 	}
 	defer r.Body.Close()
 
-	err := p.backend.ProcessVerifyNewUser(vnu)
-	if err != nil {
-		userErr, ok := err.(v1.UserError)
-		if ok {
-			url, err := url.Parse(routePrefix + v1.RouteVerifyNewUserFailure)
-			if err == nil {
-				q := url.Query()
-				q.Set("errorcode", string(userErr.ErrorCode))
-				url.RawQuery = q.Encode()
-				http.Redirect(w, r, url.String(), http.StatusMovedPermanently)
-				return
-			}
-		}
-
-		log.Errorf("handleVerifyNewUser: %v", err)
-		http.Redirect(w, r, routePrefix+v1.RouteVerifyNewUserFailure,
-			http.StatusMovedPermanently)
+	user, err := p.backend.ProcessVerifyNewUser(vnu)
+	if err != nil {
+		RespondWithError(w, r, 0, "handleLogin: ProcessVerifyNewUser %v",
+			err)
 		return
 	}
 
-	http.Redirect(w, r, routePrefix+v1.RouteVerifyNewUserSuccess,
-		http.StatusMovedPermanently)
+	util.RespondWithJSON(w, http.StatusOK,
+		v1.LoginReply{
```

Comment:
Why is `LoginReply` being returned here?

Reply:
left over stupid; fixed